dist/s_all

```diff
@@ -53,16 +53,11 @@ errchk()
     sed -e 's/^/    /' $2
     echo "#######################"
 
-    # If the test was skipped, ignore the failure.
-    if ! `grep "$1.*skipped" $2 > /dev/null`; then
+    # If the test was skipped or marked warning only, ignore the failure.
+    if [ ! `grep "$1.*skipped" $2 > /dev/null` ] && [ ! "$3" = "--warning-only" ] ; then
         errfound=1;
     fi
 
-    # Ignore errors if the script is marked warning-only
```

Comment:
I restructured this, since `errfound` is carried across multiple calls to this function, the way this is structured, and failure before a `warning-only` failure would be masked.